lib/smart_answer/calculators/state_pension_age_calculator.rb

```diff
@@ -16,6 +16,19 @@ def state_pension_date
       StatePensionDateQuery.state_pension_date(dob, gender)
     end
 
+    def approaching_pension_age?
+      difference = (state_pension_date - Date.today).to_i
+      if difference >= 0 && difference <= claim_can_start_within
+        true
+      else
+        false
+      end
+    end
+
+    def claim_can_start_within
+      (4.month.since.to_date - Date.today).to_i + 4
+    end
```

Comment:
I might be wrong, but I think this could all be written a lot more simply e.g. something like this:

```ruby
    def earliest_application_date
      state_pension_date - 4.months - 4.days
    end

    def can_apply?
      Date.today >= earliest_application_date
    end
```